### DISCUSSION TITLE
No able to assignRole when using team?

### DISCUSSION THREAD

**Discussion by clickervinod** [spatie/laravel-permission#2129] [Q&A]
Hi,
 I am not able to assign role ( $user->assignRole('customer') / $user->assignRole($role) ) when uing team getting below error
 
 `Spatie \ Permission \ Exceptions \ RoleDoesNotExist
There is no role named `customer`.`

`  SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'team_id' cannot be nul
l (SQL: insert into `model_has_roles` (`model_id`, `model_type`, `role_id`, `team_id`)
 values (1000000004, App\Models\User, 1, ?))`
 
i have allready created customer role.

Here is my rolePremission seeder

`   // Seeder
   public function run()
      {
        // Reset cached roles and permissions
       app()[\Spatie\Permission\PermissionRegistrar::class]->forgetCachedPermissions();
        Permission::create(['name' => 'view order']);
        Permission::create(['name' => 'edit order']);
        Permission::create(['name' => 'delete order']);
        Permission::create(['name' => 'create order']);
        Permission::create(['name' => 'unpublish order']);

        $role = Role::create(['name' => 'superadmin', 'team_id' => null])
            ->givePermissionTo(Permission::all());

        $role = Role::create(['name' => 'admin', 'team_id' => null])
            ->givePermissionTo(['create order', 'unpublish order']);

        $role = Role::create(['name' => 'vendor', 'team_id' => null])
            ->givePermissionTo(['create order', 'unpublish order']);

        $role = Role::create(['name' => 'salesman', 'team_id' => null])
            ->givePermissionTo(['create order', 'unpublish order']);

        $role = Role::create(['name' => 'engineer', 'team_id' => null])
            ->givePermissionTo(['create order', 'unpublish order']);

        $role = Role::create(['name' => 'customer', 'team_id' => null])
            ->givePermissionTo('edit order');

        // team user role
        $role = Role::create(['name' => 'manager'])
            ->givePermissionTo('edit order');
    }`

Using -
 "laravel/framework": "^9.13",
 "spatie/laravel-permission": "^5.5",

**Comment by erikn69** [comment]
[ISSUE_TEMPLATE BUG_REPORT](https://github.com/spatie/laravel-permission/blob/main/.github/ISSUE_TEMPLATE/bug_report.md)
Read all the info before opening a new issue
https://spatie.be/docs/laravel-permission/v5/basic-usage/teams-permissions#content-rolespermissions-assignment--removal
https://github.com/spatie/laravel-permission/discussions/2070#discussioncomment-2588991
https://github.com/spatie/laravel-permission/issues/1992#issuecomment-1013104967
https://github.com/spatie/laravel-permission/issues/1954#issuecomment-991890695
https://github.com/spatie/laravel-permission/discussions/1840#discussioncomment-1277282

**Comment by Patapoof** [comment]
Lots of people, including myself, run into this problem. I'm sorry to say that the documentation regarding this point is a mess. 
The time it took to write your passive aggressive answer would have been better employed improving the doc. Apparently, that won't come from the confused users.

**Reply by erikn69** [reply]
The time it took to write your passive aggressive answer would have been better employed improving the doc. Feel free to make a PR for better documentation

**Reply by erikn69** [reply]
I am not the author of the package, it is not my obligation to respond nor do I receive any compensation for doing so, I only responded to help, and even so people are ungrateful
https://github.com/spatie/laravel-permission/blob/67e54e2c7941d7aaa93c207549cdb7c5a44c68f5/.github/ISSUE_TEMPLATE/bug_report.md?plain=1#L10-L11

**Reply by Patapoof** [reply]
Suggesting one of those confused users, me, to write a PR is yet another aggressive answer: you know that I can't fix those things since I just told you that I was one of them.

Anyway, I wasn't coming after you, especially since you're not the author and doing this on your free time. But I am offended by the attitude and the tone: in my world, when several people express that they can't make sense of the doc, I'd probably point the finger at the doc and do something about it, I'd certainly not scold them for not reading the whole footnotes (just in case a comment somewhere would explain what the doc leaves in the shadows). 

Anyway, I'll leave the teams feature aside and will revisit when I have time to dive into what hasn't been properly documented. I guess I'll make a PR then.

**Reply by systemsolutionweb** [reply]
It seems well documented to me, do not blame others for not being able to interpret correctly
